### ISSUE TITLE
[v0.20.603] Bates fight gets stuck at the end

### ISSUE THREAD

**Issue by MagLjudik** [FullFrontalFrog/OhSoHeroTracker#147]
As reported by some users, sometimes when you finish fighting Bates, the fight does not end and Bates does not stand up. Proving player logs by one of our members: 

[Player (1).log](https://github.com/user-attachments/files/17063200/Player.1.log)
[Player-prev.log](https://github.com/user-attachments/files/17063201/Player-prev.log)

Ket is aware of this bug and is doing his best to fix it.

**Comment by FullFrontalFrog** [comment]
Here's are potential points of error:

```
NullReferenceException
  at (wrapper managed-to-native) UnityEngine.Transform.get_rotation_Injected(UnityEngine.Transform,UnityEngine.Quaternion&)
  at UnityEngine.Transform.get_rotation () [0x00000] in <7efe8e96595b432b9a58a8b0229c7ec5>:0 
  at OSH.ProjectileData.get_Velocity () [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at OSH.Bates_AttackFirePellets_Logic+<>c__DisplayClass11_0.<SpawnFirePellets>b__0 (_0G.TimeTrigger _) [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.FireEvent () [0x00038] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.Update (System.Single delta, System.Boolean updatesTimeElapsed) [0x0005a] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.Update (System.Single delta) [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeThread.UpdateTriggers () [0x0001c] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeThread.FixedUpdate () [0x00028] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at OSH.OhSoTimeManager.FixedUpdate () [0x00004] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.G.FixedUpdate () [0x00019] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
```
```
NullReferenceException
  at (wrapper managed-to-native) UnityEngine.Transform.get_rotation_Injected(UnityEngine.Transform,UnityEngine.Quaternion&)
  at UnityEngine.Transform.get_rotation () [0x00000] in <7efe8e96595b432b9a58a8b0229c7ec5>:0 
  at OSH.ProjectileData.get_Velocity () [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at OSH.Bates_AttackFirePellets_Logic+<>c__DisplayClass11_0.<SpawnFirePellets>b__0 (_0G.TimeTrigger _) [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.FireEvent () [0x00038] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.Update (System.Single delta, System.Boolean updatesTimeElapsed) [0x0005a] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeTrigger.Update (System.Single delta) [0x00000] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeThread.UpdateTriggers () [0x0001c] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.TimeThread.FixedUpdate () [0x00028] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at OSH.OhSoTimeManager.FixedUpdate () [0x00004] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
  at _0G.G.FixedUpdate () [0x00019] in <a1bb3b075e9f4772b05bb98599bc98bd>:0 
```
```
Item index -1 is not available.
```
```
Unloading 10 Unused Serialized files (Serialized files now loaded: 4)
Unloading 3898 unused Assets to reduce memory usage. Loaded Objects now: 6649.
Total: 23.333708 ms (FindLiveObjects: 0.335208 ms CreateObjectMapping: 0.267708 ms MarkObjects: 17.513209 ms  DeleteObjects: 5.217542 ms)
```
```
object manager cleanup
```
```
Thread 0x16f7cf000 may have been prematurely finalized
```

**Comment by FullFrontalFrog** [comment]
Currently unable to replicate although I've seen it happen. I'll keep trying to look into this and examine my code more closely.

**Comment by Armodillomatt12** [comment]
i've noticed that going into the lowest settings possible and beating the boss causes bates to get stuck upon being beaten
i'm using the 64 version and not the 32 version but idk if that also works in the 32 version

**Comment by FullFrontalFrog** [comment]
> i've noticed that going into the lowest settings possible and beating the boss causes bates to get stuck upon being beaten i'm using the 64 version and not the 32 version but idk if that also works in the 32 version

Settings? Which settings? Graphic quality? Do you have an error log you can provide?

**Comment by Armodillomatt12** [comment]
> > i've noticed that going into the lowest settings possible and beating the boss causes bates to get stuck upon being beaten i'm using the 64 version and not the 32 version but idk if that also works in the 32 version
> 
> Settings? Which settings? Graphic quality? Do you have an error log you can provide?

idk how to get the error log though

**Comment by FullFrontalFrog** [comment]
> > > i've noticed that going into the lowest settings possible and beating the boss causes bates to get stuck upon being beaten i'm using the 64 version and not the 32 version but idk if that also works in the 32 version
> > 
> > 
> > Settings? Which settings? Graphic quality? Do you have an error log you can provide?
> 
> idk how to get the error log though

https://wiki.ohsohero.com/error_logs

**Comment by Armodillomatt12** [comment]
[Player.log](https://github.com/user-attachments/files/17604012/Player.log)
[Player-prev.log](https://github.com/user-attachments/files/17604013/Player-prev.log)
here's the files
also it seems to happen when both the health meter and the armor meter are at 0, which means having sex with bates as the finishing blow breaks the game.
try doing that to see if that's the reason behind the bug

**Comment by FullFrontalFrog** [comment]
> [Player.log](https://github.com/user-attachments/files/17604012/Player.log) [Player-prev.log](https://github.com/user-attachments/files/17604013/Player-prev.log) here's the files also it seems to happen when both the health meter and the armor meter are at 0, which means having sex with bates as the finishing blow breaks the game. try doing that to see if that's the reason behind the bug

Thank you for providing these! It's difficult to say if this will lead to a fix, but it helps~

**Comment by FullFrontalFrog** [comment]
I was finally able to replicate the issue and I believe I found one or more causes of the problem, which is related to improper AI routine/state management when Bates is defeated. It was difficult to find because it was not generating any errors or exceptions in the logs. Fortunately, I have a possible fix in place! I'll move forward with this for now and have some additional testing done soon.

**Comment by MagLjudik** [comment]
I fought Bates a few times and did not find any bugs, and not this one specifically.
So we can consider this fixed as of 0.20.700
